### ISSUE TITLE
[BUG] Popup: Android screen reader accessibility

### ISSUE THREAD

**Issue by IeuanWalker** [CommunityToolkit/Maui#2958]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

Currently on Android, when using the screen reader, the popup view itself takes all the focus, and it's impossible to focus on non-interactive content of the popup _(i.e. labels aren't accessible, but buttons are)_

### Expected Behavior

The popup view shouldn't be focusable, and all inner elements should be accessible

### Steps To Reproduce

Turn on the screen reader and use the sample app

### Link to public reproduction project repository

sample app

### Environment

```markdown
- .NET MAUI CommunityToolkit:
- OS:
- .NET MAUI:
```
